### ISSUE TITLE
frontend: remove superagent and use XHR for the post request

### ISSUE THREAD

**Issue by XhmikosR** [emedvedev/slackin-extended#86]

**Comment by joelpurra** [comment]
I agree, in particular since superagent is given special path treatment in `index.js`.

- https://github.com/emedvedev/slackin-extended/blob/v2.1.0/lib/index.js#L128

On my system `/assets/superagent.js` returns HTTP error 404 and breaks `slackin-extended`. This is due to (changes) in how `npm install` places dependencies, which _may_ be optimized between packages in the same `package.json`. I have a local hot-fix in place for now.

- https://slackin.joelpurra.com/

<details>
<summary>Versions</summary>

Am running slightly older versions due to `apt` on Debian 10 Buster.

```shell
node --version
```

```text
v10.15.2
```

```shell
npm --version
```

```text
6.14.4
```

</details>

<details>
<summary>Paths</summary>

```shell
npm ls superagent
```

```text
slackin.joelpurra.com@1.0.0 .../slackin.joelpurra.com/public_html
└─┬ slackin-extended@2.1.0
  └── superagent@3.8.3 
```

```shell
ls --format=long --directory ./node_modules/{slackin-extended,superagent}
```

```text
drwxr-xr-x 1 root root 116 Apr  8 16:56 ./node_modules/slackin-extended
drwxr-xr-x 1 root root 282 Apr  8 16:56 ./node_modules/superagent
```

</details>

Perhaps something like `resolve-pkg` could be used in the short-term, but replacing `superagent` with something else might be a better long-term solution.

- https://github.com/sindresorhus/resolve-pkg

**Comment by XhmikosR** [comment]
My suggestion would be to just drop the dependency but if it's easier feel free to submit a PR with resolve-pkg. You should make a new issue about it BTW

**Comment by joelpurra** [comment]
@XhmikosR: no rush for me, as I have a local hotfix in place. Will await a proper solution before upgrading.

**Comment by XhmikosR** [comment]
As you wish, but this issue is not about the issue you have, which is a bug I introduced for what is worth. I'll try to have a look in the next days but this deserves its own issue.

**Comment by joelpurra** [comment]
@XhmikosR: true, it's not the same issue. See #185.
